### PR TITLE
add support for running in the hardened cluster

### PR DESCRIPTION
**Pull request by jiaqiluo**
issue https://github.com/rancher/rancher/issues/34188

a corresponding PR in rancher/rancher: https://github.com/rancher/rancher/pull/34286

Note:
This PR is already tested in a hardened cluster

TODO:

- [ ] cut a new tag

### CODE REVIEW

File: charts/rancher-webhook/templates/pre-delete-hook-cluster-role.yaml [hunk 1]
```diff
@@ -13,4 +13,10 @@ rules:
     resources: [ "mutatingwebhookconfigurations" ]
     verbs: [ "delete" ]
     resourceNames: [ "rancher.cattle.io"]
+  - apiGroups: [""]
+    resources: ["serviceaccounts"]
+    verbs: ["get"]
+  - apiGroups: ["policy"]
+    resources: ["podsecuritypolicies"]
+    verbs:     ["use"]
```
Comment (StrongMonkey) [inline charts/rancher-webhook/templates/pre-delete-hook-cluster-role.yaml:21]: Do you have to specify the name of the PSP to use with `resourceName`?

Comment (jiaqiluo) [reply charts/rancher-webhook/templates/pre-delete-hook-cluster-role.yaml]: It is optional but I just added it.

**Review by StrongMonkey — APPROVED** [review]

**Review by aiyengar2 — APPROVED** [review]
LGTM

**Review by cbron — APPROVED** [review]

File: charts/rancher-webhook/templates/pre-delete-hook-cluster-role.yaml (hunk 1 above)
Comment (cbron) [reply charts/rancher-webhook/templates/pre-delete-hook-cluster-role.yaml]: Didn't realize it was optional.

**Pull request merged**